### DISCUSSION TITLE
Plugin Websockets v2 use-case example?

### DISCUSSION THREAD

**Discussion by jswhisperer** [tauri-apps/tauri#10917] [Q&A] [answered]
[websockets](https://v2.tauri.app/plugin/websocket/#_top)
I'm just wondering this might apply to a few plugins could someone explain when would you use this versus a javascript / webview websocket. What would be the use case or advantage?

**Comment by jswhisperer** [comment]
I actually think I was being silly by looking at this from a javascript developer perspective.  

**Please let me know this if this is correct and the only use case:**
To be able to interact with a websocket connection directly from Rust code

**Comment by FabianLars** [comment, marked as answer]
The main reason is the same as the reason to use the http plugin. It works even when the browser's apis fail due to things like CORS. Also, some people may just prefer to have things like these implemented in Rust (even though it may be a bit slower due to the IPC) 🤷 

Also, so far a pretty nieche feature, but you can specifiy a custom TLS connector (in rust).

> To be able to interact with a websocket connection directly from Rust code

Nope, the plugin doesn't have a Rust api yet, other than configuration.
